src/mpyl/cli/commands/build/mpyl.py

```diff
@@ -126,18 +125,22 @@ def find_build_set(repo: Repository, changes_in_branch: list[Revision], build_al
 
 
 def run_build(accumulator: RunResult, executor: Steps, reporter: Optional[Reporter] = None, dry_run: bool = True):
-    for stage, projects in accumulator.run_plan.items():
-        for proj in projects:
-            result = executor.execute(stage, proj, dry_run)
-            accumulator.append(result)
-            if reporter:
-                reporter.send_report(accumulator)
-
-            if not result.output.success and stage == Stage.DEPLOY:
-                logging.warning(f'Deployment failed for {proj.name}')
+    try:
+        for stage, projects in accumulator.run_plan.items():
+            for proj in projects:
+                result = executor.execute(stage, proj, dry_run)
+                accumulator.append(result)
+                if reporter:
+                    reporter.send_report(accumulator)
+
+                if not result.output.success and stage == Stage.DEPLOY:
```

Comment:
You probably noticed already, but we made sure other stages now only fail after finishing all the projects. Might be something we have to make configurable as well